http.c

```diff
@@ -63,7 +67,11 @@ void inject_header(buffer_t *buffer, struct phr_header *headers, size_t num_head
 	char *new_header_pos = headers[num_headers - 1].value + headers[num_headers - 1].value_len + 2;
 
 	// shift the rest of the buffer
-	buffer_grow(buffer, new_header_len); // resize the buffer if necessary
+	if (!buffer_grow(buffer, new_header_len)) // resize the buffer if necessary
+	{
+		pr_err("could not inject header: could not allocate memory");
```

Comment:
I would distinguish these error messages line 42 and 72.

Reply:
IMHO it does not matter in case of a memory allocation error where it happened.